### DISCUSSION TITLE
Error in kernel patch-file

### DISCUSSION THREAD

**Discussion by joerg-wille** [Freetz-NG/freetz-ng#925] [Allgemein]
Trying to build an image for 7520.
The build gets terminated when the kernel gets patched for AVM.

That's the make log:
Using kernel version: 4.4.271
rm -f -r source/kernel/ref-ipq40xx-7530_07.56
mkdir -p source/kernel/ref-ipq40xx-7530_07.56/linux-4.4
---> kernel ...
preparing ... patching ...
#vanilla to avm patch: dl/kernel_4.4.271-7530_07.56-ADv1.patch.xz
ERROR: modpatch: Error in patch-file dl/kernel_4.4.271-7530_07.56-ADv1.patch.xz
make[1]: *** [make/kernel/kernel.mk:55: source/kernel/ref-ipq40xx-7530_07.56/.unpacked] Error 2
make: *** [Makefile:47: envira] Terminated

The image, which gets downloaded from AVM is "FRITZ.Box_7520-07.57.image"

That's the box juis_boxinfo.xml:
<q:Name>FRITZ!Box 7520 (UI)</q:Name>
<q:HW>247</q:HW>
<q:Major>175</q:Major>
<q:Minor>7</q:Minor>
<q:Patch>57</q:Patch>
<q:Buildnumber>107821</q:Buildnumber>
<q:Buildtype>1</q:Buildtype>
<q:OEM>1und1</q:OEM>
<q:Lang>de</q:Lang>
<q:Country>049</q:Country>

**Comment by joerg-wille** [comment]
The error has not been fixed, but I am not stuck any more.
I have been using docker on Mac using this image:https://github.com/pfichtner/pfichtner-freetz
This is based on Ubuntu 20. I did update the system and checked all prerequisites.
The freetz-ng repo was checked out on the host and passed to docker.

I now switches to use VMWare using Ubuntu Server 23.
With following steps I succeeded building an image:
1. Install Ubuntu Server 23 in VMWare in custom VM (100GB)
Use: ubuntu-23.10-live-server-amd64.iso

2. Prepare
ssh freetz@172.16.199.133
sudo apt update && sudo apt upgrade -y && apt-get -y dist-upgrade
sudo apt-get install open-vm-tools git
git clone https://github.com/Freetz-NG/freetz-ng.git && cd freetz-ng
mkdir tools-host && cd tools-host && wget https://ftp.gnu.org/gnu/gmp/gmp-4.2.4.tar.bz2 && cd ..
./tools/prerequisites install -y Ubuntu23
sudo reboot

3. Build
cd freetz-ng
make menuconfig
(e.g. set 7520, exit and save)
make

**Reply by pfichtner** [reply]
@joerg-wille you say you have a bug under Docker that you don't have under VMWare? If so, please open an issue, preferably with your config.
https://github.com/pfichtner/pfichtner-freetz/issues

works for me

```
 rm .config && make menuconfig && make

[...]
STEP 3: PACK/SIGN
  checking for left over version-control-system files
  integrate freetz info file into image
packing var.tar
  checking signature key files
    creating private signature key file
    creating public signature key file
    adding public signature key file
creating filesystem image (SquashFS4-xz)
  SquashFS block size: 64 kB (65536 bytes)
copying kernel image
  kernel image size: 3.1 MB, max 4.0 MB, free 0.9 MB (900608 bytes)
copying filesystem image
  filesystem image size: 28.2 MB, max 44.0 MB, free 15.8 MB (16515072 bytes)
adding checksum to kernel.image
adding checksum to filesystem.image
packing images/7520_07.57.all_freetz-ng-23069M-be072c1ff_20240126-215309.image
  packed image file size: 36.1 MB (37898240 bytes)
signing packed .image file
  signed image file size: 36.1 MB (37898240 bytes)
source firmware: 7520_de-es-fr-it-nl-pl 175.07.57 rev107821 {ALL} [MOVE21NL1] (01.09.2023 23:52:32) 
  source image file size: 35.6 MB (37355520 bytes)
done.

git rev-list --max-count=1 HEAD
be072c1ffec695792cb36f3cc539d43576206c58
```

**Comment by fda77** [comment]
imho only way to cause this error is an alterned file after download
